projects/RabbitMQ.Client/client/impl/Frame.cs

```diff
@@ -75,14 +65,17 @@ internal static class Method
              * +----------+-----------+-----------+ */
             public const int FrameSize = BaseFrameSize + 2 + 2;
 
-            public static int WriteTo(Span<byte> span, ushort channel, MethodBase method)
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static int WriteTo<T>(Span<byte> span, ushort channel, T method) where T : MethodBase
             {
                 const int StartClassId = StartPayload;
-                const int StartMethodArguments = StartPayload + 4;
+                const int StartMethodArguments = StartClassId + 4;
 
+                int payloadLength = method.WriteArgumentsTo(span.Slice(StartMethodArguments)) + 4;
+                NetworkOrderSerializer.WriteUInt64(span, ((ulong)Constants.FrameMethod << 56) | ((ulong)channel << 40) | ((ulong)payloadLength << 8));
```

Comment:
I assume this is repeated as it is not inlineable if it is in its own method?

Reply:
It's just a one-liner, so not worried about it.